### DISCUSSION TITLE
Use glob pattern with add_entity_to_area

### DISCUSSION THREAD

**Discussion by anoblet** [frenck/spook#298] [Q&A] [answered]
I'm a YAML person and have created a fair amount of helpers to help with everything. It would be nice to have a way to automatically, or at least quickly add all entities that match a certain pattern (`kitchen_*`) to an area so that even if they aren't defined in a specific dashboard, when using the overview tab, they are correctly organized.

**Comment by frenck** [comment]
To be honest, allowing such patterns in service calls don't make much sense to me. That said, you can already do this using a template, right?

**Reply by anoblet** [reply]
I've done this using HA CLI for devices, though there isn't a way to do this for entities. Could you perhaps point me in the right direction on how to use templates to accomplish this?

**Reply by frenck** [reply, marked as answer]
![CleanShot 2023-08-10 at 23 11 01](https://github.com/frenck/spook/assets/195327/c47ffd44-e683-47f5-9e49-b8f35d920be6)

Template:
```jinja2
{{ 
   states
   | selectattr('entity_id', 'search', '^light.living_*')
   | map(attribute="entity_id")
   | list
}}
```

Combined in a service call:

```yaml
- service: homeassistant.add_entity_to_area
  data:
    entity_id: >
      {{ 
         states
         | selectattr('entity_id', 'search', '^light.living_*')
         | map(attribute="entity_id")
         | list
      }}
```

**Reply by frenck** [reply]
Alternatively, a little cleaner even maybe:

```
{{ 
   states
   | map(attribute="entity_id")
   | select('search', '^light.living_*')
   | list
}}
```

**Reply by anoblet** [reply]
You're awesome! Thank you so much for what you do! :)

**Reply by frenck** [reply]
☺️ Thank you, no problem 👍
